gramps/gen/const.py

```diff
@@ -104,10 +105,20 @@
     else:
         USER_DATA = os.path.join(USER_HOME, "AppData", "Roaming", "gramps")
     USER_CONFIG = USER_DATA
+    # Migrate data from AppData\Local to AppData\Roaming on Windows.
+    OLD_HOME = os.path.join(USER_HOME, "AppData", "Local", "gramps")
+    if os.path.exists(OLD_HOME) and not os.path.exists(USER_DATA):
+        shutil.move(OLD_HOME, USER_DATA)
 else:
     USER_HOME = get_env_var("HOME")
     USER_DATA = os.path.join(GLib.get_user_data_dir(), "gramps")
     USER_CONFIG = os.path.join(GLib.get_user_config_dir(), "gramps")
+    # Copy database directory into the XDG data directory.
+    OLD_HOME = os.path.join(USER_HOME, ".gramps")
+    if os.path.exists(OLD_HOME) and not os.path.exists(USER_DATA):
+        shutil.copytree(
+            os.path.join(OLD_HOME, "grampsdb"), os.path.join(USER_DATA, "grampsdb")
+        )
```

Comment:
What about USER_CONFIG?

Reply:
We probably only need the `gramps.ini` file.

I also need to check that the `grampsdb` directory exists before attempting to copy it.

Reply:
> We probably only need the gramps.ini file.

I think we just need for [config.py](https://github.com/gramps-project/gramps/blob/master/gramps/gen/config.py#L410) to check the old location.

Reply:
I'll look into it.  Using existing code and the same upgrade path would be a better option.

Reply:
I've added a check for the old path to the config update code.